internal/overlord/servstate/manager.go

```diff
@@ -425,3 +445,19 @@ func (m *ServiceManager) SendSignal(services []string, signal string) error {
 	}
 	return nil
 }
+
+// CheckFailure response to a health check failure. If the given check name is
```

Comment:
"CheckFailure" reads too much like "check this failure" (verb) instead of "a check has failed" (noun). Will be solved by the earlier point, though, assuming there's some agreement around it.